src/connectedk8s/azext_connectedk8s/_utils.py

```diff
@@ -0,0 +1,110 @@
+                                        summary='Provided location is not supported for creating connected clusters')
+                raise CLIError("Connected cluster resource creation is supported only in the following locations: " +
+                               ', '.join(map(str, rp_locations)) +
+                               ". Use the --location flag to specify one of these locations.")
+            break
+
+
+def get_chart_path(registry_path, kube_config, kube_context):
+    # Pulling helm chart from registry
+    os.environ['HELM_EXPERIMENTAL_OCI'] = '1'
+    pull_helm_chart(registry_path, kube_config, kube_context)
+
+    # Exporting helm chart
+    chart_export_path = os.path.join(os.path.expanduser('~'), '.azure', 'AzureArcCharts')
+    export_helm_chart(registry_path, chart_export_path, kube_config, kube_context)
+    # Helm Install
+    helm_chart_path = os.path.join(chart_export_path, 'azure-arc-k8sagents')
+    chart_path = os.getenv('HELMCHART') if os.getenv('HELMCHART') else helm_chart_path
+    return chart_path
+
+
+def pull_helm_chart(registry_path, kube_config, kube_context):
+    cmd_helm_chart_pull = ["helm", "chart", "pull", registry_path, "--kubeconfig", kube_config]
+    if kube_context:
+        cmd_helm_chart_pull.extend(["--kube-context", kube_context])
+    response_helm_chart_pull = subprocess.Popen(cmd_helm_chart_pull, stdout=PIPE, stderr=PIPE)
+    _, error_helm_chart_pull = response_helm_chart_pull.communicate()
+    if response_helm_chart_pull.returncode != 0:
+        telemetry.set_exception(exception=error_helm_chart_pull.decode("ascii"), fault_type=consts.Pull_HelmChart_Fault_Type,
+                                summary='Unable to pull helm chart from the registry')
+        raise CLIError("Unable to pull helm chart from the registry '{}': ".format(registry_path) + error_helm_chart_pull.decode("ascii"))
+
+
+def export_helm_chart(registry_path, chart_export_path, kube_config, kube_context):
+    chart_export_path = os.path.join(os.path.expanduser('~'), '.azure', 'AzureArcCharts')
+    cmd_helm_chart_export = ["helm", "chart", "export", registry_path, "--destination", chart_export_path, "--kubeconfig", kube_config]
+    if kube_context:
+        cmd_helm_chart_export.extend(["--kube-context", kube_context])
+    response_helm_chart_export = subprocess.Popen(cmd_helm_chart_export, stdout=PIPE, stderr=PIPE)
+    _, error_helm_chart_export = response_helm_chart_export.communicate()
+    if response_helm_chart_export.returncode != 0:
+        telemetry.set_exception(exception=error_helm_chart_export.decode("ascii"), fault_type=consts.Export_HelmChart_Fault_Type,
+                                summary='Unable to export helm chart from the registry')
+        raise CLIError("Unable to export helm chart from the registry '{}': ".format(registry_path) + error_helm_chart_export.decode("ascii"))
+
+
+def add_helm_repo(kube_config, kube_context):
+    if os.getenv('HELMREPONAME') and os.getenv('HELMREPOURL'):
```

Comment:
I think we shoud add this if condition in the custom.py file where we call this function